CoursesManager.UI/DataAccess/StudentDataAccess.cs

```diff
@@ -83,10 +83,7 @@ public void Add(Student student)
                     throw new ArgumentNullException(nameof(student.Address), "Student address cannot be null.");
                 }
 
-                _addressDataAccess.Add(student.Address);
-
-                // Get the newly created address ID
-                int addressId = _addressDataAccess.GetLastInsertedId();
+                int addressId = _addressDataAccess.Add(student.Address);
```

Comment:
Ah hierom dus. Misschien beter gewoon een int en er idd voor zorgen dat add procedures een int out'en? Beter dan dat ding van shadi waar je getlastid doet. Was dat je plan?

Reply:
Ja ik heb de functie in de DataLayer nu zo gemakt dat hij false returned als het niet goed gaat en anders de lastInsertedId returned